receiver/k8sclusterreceiver/internal/hierarchicalresourcequota/hierarchicalresourcequotas.go

```diff
@@ -0,0 +1,44 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package hierarchicalresourcequota // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/internal/hierarchicalresourcequota"
+
+import (
+	"strings"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/internal/metadata"
+)
+
+func RecordMetrics(mb *metadata.MetricsBuilder, hrq *unstructured.Unstructured, ts pcommon.Timestamp) {
+	name, _, _ := unstructured.NestedString(hrq.Object, "metadata", "name")
+	uid, _, _ := unstructured.NestedString(hrq.Object, "metadata", "uid")
+	namespace, _, _ := unstructured.NestedString(hrq.Object, "metadata", "namespace")
+	statusHard, _, _ := unstructured.NestedMap(hrq.Object, "status", "hard")
+	for k, v := range statusHard {
+		val := parseQuantityValue(k, v.(string))
+		mb.RecordK8sHierarchicalResourceQuotaHardLimitDataPoint(ts, val, k)
+	}
+	statusUsed, _, _ := unstructured.NestedMap(hrq.Object, "status", "used")
+	for k, v := range statusUsed {
+		val := parseQuantityValue(k, v.(string))
+		mb.RecordK8sHierarchicalResourceQuotaUsedDataPoint(ts, val, k)
+	}
```

Comment:
We should do some more error/"not found" handling here. If we don't get a value we expect we shouldn't set the resource attributes below.  And if we can't set the resource attribute should we still set the metric?

Reply:
I see. I'll fix them. Thank you for your feedback.

Reply:
Uid and namespace properties are optional. so, we set the metric if these properties is not found.
https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L158
https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L144

Reply:
I fixed to handle errors.